### PR TITLE
MINOR: Upgrade gradle plugins and test libraries for Java 14 support

### PR DESCRIPTION
**Pull request by ijuma**
Also:
* Remove deprecated `=` in resolutionStrategy.
* Replace `AES/GCM/PKCS5Padding` with `AES/GCM/NoPadding`
in `PasswordEncoderTest`. The former is invalid and JDK 14 rejects it,
see https://bugs.openjdk.java.net/browse/JDK-8229043.

With these changes, the build works with Java 14 and Scala 2.12. The
same will apply to Scala 2.13 when Scala 2.13.2 is released (should
happen within 1-2 weeks).

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
retest this please

**Review by omkreddy — APPROVED** [review]
@ijuma Thanks for the PR. LGTM if tests pass.

File: jenkins.sh [hunk 1]
```diff
@@ -29,12 +29,17 @@
...
-# Verify that Kafka Streams archetype compiles
 if [ $JAVA_HOME = "/home/jenkins/tools/java/latest11" ] ; then
   echo "Skipping Kafka Streams archetype test for Java 11"
   exit 0
 fi
 
+if [ $JAVA_HOME = "/home/jenkins/tools/java/latest14" ] ; then
+  echo "Skipping Kafka Streams archetype test for Java 14"
+  exit 0
+fi
```
Comment (mjsax) [inline jenkins.sh:40]: LGTM

**Comment by ijuma** [comment]
2 flaky tests, one in each job:
* org.apache.kafka.streams.integration.EOSUncleanShutdownIntegrationTest.shouldWorkWithUncleanShutdownWipeOutStateStore[exactly_once_beta]
* kafka.integration.MetricsDuringTopicCreationDeletionTest.testMetricsDuringTopicCreateDelete

**Pull request merged**